compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -427,6 +427,17 @@ class ValuePropagation : public TR::Optimization
    void invalidateParmConstraintsIfNeeded(TR::Node *node, TR::VPConstraint *constraint);
    void checkTypeRelationship(TR::VPConstraint *lhs, TR::VPConstraint *rhs, int32_t &value, bool isInstanceOf, bool isCheckCast);
    TR_YesNoMaybe isCastClassObject(TR::VPClassType *type);
+
+   /**
+    * Determine whether the component type of an array is, or might be, a value
+    * type.
+    * \param arrayConstraint The \ref TR::VPConstraint type constraint for the array reference
+    * \returns \c TR_yes if the array's component type is definitely a value type;\n
+    *          \c TR_no if it is definitely not a value type; or\n
+    *          \c TR_maybe otherwise.
+    */
+   TR_YesNoMaybe isArrayCompTypeValueType(TR::VPConstraint *arrayConstraint);
```

Comment:
I guess I should have declared this method to be `virtual` in order to allow for [this potential future rearrangement of code](https://github.com/eclipse/omr/pull/5904/#issuecomment-815846493).

Reply:
I wonder if it's worth of making `isArrayCompTypeValueType` more generic that takes ` TR_OpaqueClassBlock *arrayComponentClass` as an argument and it could be put in `OMRClassEnv` as where `isValueTypeClass` is declared.   
https://github.com/eclipse/omr/blob/e0b3f5c8bd4d23d24c1770e4762b710c63fa4568/compiler/env/OMRClassEnv.hpp#L88.

Reply:
Added `virtual` modifier to declaration of `isArrayCompTypeValueType` in https://github.com/eclipse/omr/pull/5904/commits/02aa006865b25bbd75d7411a8b5a2facb65cfc15